src/rp2/abstract_transaction.py

```diff
@@ -39,7 +39,9 @@ def __init__(
         self.__timestamp: datetime = configuration.type_check_timestamp_from_string("timestamp", timestamp)
         self.__transaction_type: TransactionType = TransactionType.type_check_from_string("transaction_type", transaction_type)
         self.__spot_price: RP2Decimal = configuration.type_check_positive_decimal("spot_price", spot_price)
-        self.__internal_id: int = configuration.type_check_internal_id("internal_id", internal_id) if internal_id is not None else id(self)
+        # TODO: the fallback case does not semantically match "row", make non-optional  # pylint: disable=fixme
```

Comment:
Could you elaborate a bit on this comment? I'm not clear what "fallback case" this refers to.

Reply:
Updated in newest commits